website/routes.py

```diff
@@ -765,6 +765,12 @@ def make_url_map(app):
         ], 'get', project_views.node.node_registrations,
             OsfWebRenderer('project/registrations.mako')),
 
+        Rule([
+            '/project/<pid>/retract_registration/',
```

Comment:
I'm thinking of something like the Thoughtworks post on [REST API Design Resource Modeling](http://www.thoughtworks.com/insights/blog/rest-api-design-resource-modeling).

```
An alternate approach that addresses the CRUD concern is to design the API around the resources that are based on the business processes and domain events. For example, to update an existing bank customer’s address, a POST request can be made to “ChangeOfAddress” resource.
```
The idea of the essay is to move towards REST without PUT, which may be a bit far for right now, but the idea is you are getting the retraction information/form/page, and your are POSTing (or PUTting) the retraction.  Regardless of the greater goal of REST without PUT, I still like the idea of a vocabulary of verbs acting on nouns, rather than verbs acting on other verbs.